src/lib/optional-modules.ts

```diff
@@ -0,0 +1,298 @@
+import { Command } from '../types';
+
+import { clear } from 'console';
+import * as util from './util';
+
+import * as files from './files';
+
+import inquirer from 'inquirer';
+import { Group } from '../types/cli';
+
+import { concat, flatten } from 'lodash';
+import { ADD_ACTION, ADD_GROUP, LIST_ACTION } from '../constants/constants';
+
+import { CLI_DESCRIPTION } from '..';
+
+import chalk from 'chalk';
+
+import * as MODULE_NEW from '../modules/new';
+import { readMainConfig } from './files';
+
+/**
+ * Description - Accepts a Feature Group name, prompts user to select a module from that group
+ * then calls the promptQuestionByGroup() function to add the mdule to the project
+ *
+ * @param groupName - Name of Feature Group
+ */
+async function handleAddGroup(groupName: string) {
+  const group = util.getFeatureGroupByName(groupName);
+
+  if (group !== undefined) {
+
+    // Prompt User to select Feature(s)
+    const selectedModules = await promptQuestionByGroup(group);
+    for (const module of selectedModules) {
+      await addPlugin(module);
+    }
+  }
+}
+
+/**
+ * Description - Accepts a feature group and Prompts the user with its question
+ * and returns an array of the selected modules
+ * @param group - Group to prompt by
+ */
+async function promptQuestionByGroup(group: Group) {
+  const multiple = '(multiple)';
+  const single = '(single)';
+  const result = Array();
+  const confirmBtn = group.isMultipleChoice ? 'Press Enter to continue' : '';
+
+  const selectionBtn = group.isMultipleChoice ? `${chalk.blueBright('<space>')} to select,${chalk.blueBright('<a>')} to toggle all, ${chalk.blueBright('<i>')} to invert selection` : 'Enter';
+
+  const instructions = chalk.yellow(`Use arrow keys to navigate. Press ${selectionBtn} to select an option. ${confirmBtn}`);
+
+
+  const question: inquirer.QuestionCollection = {
+    type: group.isMultipleChoice ? 'checkbox' : 'list',
+    name: 'feature',
+    prefix: '',
+    message: `\n \n${instructions} \n${group.question} ${group.isMultipleChoice ? multiple : single}`,
+    choices: ['None', ...group.plugins]
+  };
+
+  const selected = await inquirer.prompt(question);
+
+  result.push(selected.feature);
+
+  return flatten(result) as string[];
+}
+
+
+/**
+ * Description - Accepts the name of an optional module and adds the module to project
+ * @param plugin - name of module
+ */
+async function addPlugin(plugin: string) {
+  if (isPlugin(plugin) || isDefaultPlugin(plugin)) {
+    await MODULE_NEW.run(
+      {
+        action: ADD_ACTION,
+        feature: plugin,
+        options: [],
+      },
+      CLI_DESCRIPTION);
+  }
+}
+
+/**
+ * Description - Checks if the feature inputted is an optional feature
+ * @param feature - name of feature
+ */
+function isPlugin(plugin: string): boolean {
+  let found;
+  const plugins = readMainConfig()?.plugins;
```

Comment:
Unsure  about this... The reasons behind having '?' at the end of a property is because it could be undefined... So by way this is implemented you are 100% sure that it wont be undefined? If not then checking if it is undefined and then doing something is best.